packages/xdl/src/project/createBundlesAsync.ts

```diff
@@ -71,18 +84,15 @@ export async function createBundlesAsync(
     }
   }
 
-  const isLegacy = isLegacyImportsEnabled(
-    getConfig(projectRoot, { skipSDKVersionRequirement: true }).exp
-  );
-  // If not legacy, delete the target option to prevent warnings from being thrown.
-  if (!isLegacy) {
-    delete publishOptions.target;
-  }
+  const config = getConfig(projectRoot, { skipSDKVersionRequirement: true });
+  const isLegacy = isLegacyImportsEnabled(config.exp);
   const platforms: Platform[] = ['android', 'ios'];
   const [android, ios] = await bundleAsync(
     projectRoot,
+    publishOptions.target ?? getDefaultTarget(projectRoot),
```

Comment:
ideally i'd like to avoid adding any new uses for the target flag. what we have been doing in similar cases to this instead is we:

- check for the existence of the ios/android project
- use that to read the information we need if it's available
- otherwise we read from expo app config.

Reply:
based on this, `expo publish --target` and `expo export --target` will not be supported, at least for hermes support.
is this right?

Reply:
correct!

Reply:
if anything is unclear (i imagine it is) then let's chat about it tomorrow :)

Reply:
thanks for the briefly explanation. 
that would be fine if we don't support the `--target` which caused confusion and make the code clearer.

just thinking about app config as single source of truth for a long time.
as mentioned, the assumption for user to setup native project properly will be a problem.
an idea is that we could show a warning during export/publish if finding inconsistencies between native project and app config.
e.g. app.json setups as hermes but gradle.properties setup for jsc, we could show a warning prompt.

let's talk about this later.

Reply:
introduced the `maybeInconsistentEngineAsync()` to show the error prompt.
also updated the PR description with screenshot.
please help to review again. thanks.